contracts/wallet.sol

```diff
@@ -294,9 +294,8 @@ contract SpendLimit is Controllable, Ownable {
     /// @dev Update available spend limit based on the daily reset.
     function _updateSpendAvailable() internal {
         if (now > _spendLimitDay.add(24 hours)) {
-            // Advance the current day by how many days have passed.
-            uint extraDays = now.sub(_spendLimitDay).div(24 hours);
-            _spendLimitDay = _spendLimitDay.add(extraDays.mul(24 hours));
+            // Advance the current timestamp.
+            _spendLimitDay = now;
```

Comment:
This looks sane, thank you ... I guess we should port this, and the tests we now have added to the 2.X branch. @i-stam

Reply:
I 'll do it